aws-geo-location/src/test/java/com/amplifyframework/geo/location/tracking/UploadWorkerTest.kt

```diff
@@ -0,0 +1,257 @@
+    }
+
+    @Test
+    fun `batches locations by time`() {
+        // setup: 5 locations, with time-based batching.  the third location barely does not exceed the time threshold,
+        // so we start a new batch after processing location 3.  locations 4 and 5 are submitted in their own batch.
+        val now = Instant.now()
+        val options = GeoTrackingSessionOptions.builder()
+            .withBatchingOptions(BatchingOptions.secondsElapsed(7))
+            .build()
+        UploadWorker.options = options
+        val locations = (0 until 5).map {
+            LocationEntity(it.toLong(), id, tracker, now.plusMillis(it * 3000L), lat + it * .05, long)
+        }
+        coEvery {mockLocationDao.getAll()}.returns(locations)
+
+        runBlocking {
+            val result = uploadWorker.doWork()
+            assertTrue(result.outputData.size() == 0) // success!
+            coVerify { mockLocationDao.getAll() }
+            coVerify { mockGeoService.updateLocations(id, withArg {
+                when(it.size) {
+                    3 -> {
+                        // batch 1
+                        assertEq(it[0], position(lat + 0 * .05, long))
+                        assertEq(it[1], position(lat + 1 * .05, long))
+                        assertEq(it[2], position(lat + 2 * .05, long))
+                    }
+                    2 -> {
+                        // batch 2
+                        assertEq(it[0], position(lat + 3 * .05, long))
+                        assertEq(it[1], position(lat + 4 * .05, long))
+                    }
+                    else -> assert(false)
+                }
+            }, any()) }
+            coVerify { mockLocationDao.removeAll(withArg {
+                assertTrue(it.size == 5)
+                for (i in 0 until 5) {
+                    val position = (it as List<LocationEntity>)[i]
+                    assertTrue(position.deviceId == id)
+                    assertTrue(position.tracker == tracker)
+                    assertEquals(position.latitude, lat + i * .05, 0.001)
+                    assertTrue(position.longitude == long)
+                }
+            }) }
+        }
+        confirmVerified(mockGeoService, mockLocationDao)
+    }
+
+    @Test
+    fun `supports proxy delegate`() {
+        val mockProxyDelegate: GeoTrackingSessionOptions.LocationProxyDelegate = mockk()
+        coEvery {mockProxyDelegate.updatePositions(any())}.just(Runs)
+        UploadWorker.options = GeoTrackingSessionOptions.builder().withProxyDelegate(
+            mockProxyDelegate
+        ).build()
+        val now = Instant.now()
+        val locations = (0 until 5).map {
+            LocationEntity(it.toLong(), id, tracker, now.plusMillis(it * 3000L), lat + it * .05, long)
+        }
+        coEvery {mockLocationDao.getAll()}.returns(locations)
+
+        runBlocking {
+            val result = uploadWorker.doWork()
+            assertTrue(result.outputData.size() == 0) // success!
```

Comment:
Using `assertTrue` isn't great since a failure simply yields a message like "expected false to be true". Let's at least use `assertEquals` until we get a better assertion library in place.